numpy/core/src/multiarray/descriptor.c

```diff
@@ -727,7 +720,7 @@ _convert_from_commastring(PyObject *obj, int align)
     PyObject *_numpy_internal;
 
     if (!PyUnicode_Check(obj)) {
-        _report_generic_error();
+        PyErr_SetString(PyExc_TypeError, "Object needs to be a str");
```

Comment:
This one is unreachable, so perhaps just change this to an `assert`

Reply:
Done.